subsys/storage/flash_map/flash_map_default.c

```diff
@@ -18,21 +18,14 @@
 	COND_CODE_1(DT_NODE_HAS_COMPAT(GPARENT(part), soc_nv_flash), \
 		    (DT_LABEL(DT_PARENT(GPARENT(part)))), (_else_code))
```

Comment:
I like the refactoring, but there's a lot of gratuitous indirection defining macros that are themselves used in only one location.  I believe the change can be simplified to:
```
diff --git a/subsys/storage/flash_map/flash_map_default.c b/subsys/storage/flash_map/flash_map_default.c
index d7bfd04072..3ab39e1e3f 100644
--- a/subsys/storage/flash_map/flash_map_default.c
+++ b/subsys/storage/flash_map/flash_map_default.c
@@ -13,26 +13,11 @@
 /* Get the grand parent of a node */
 #define GPARENT(node_id) DT_PARENT(DT_PARENT(node_id))
 
-/* if 'soc-nv-flash' return controller label or punt to _else_code */
-#define IS_SOC_NV_FLASH(part, _else_code) \
-       COND_CODE_1(DT_NODE_HAS_COMPAT(GPARENT(part), soc_nv_flash), \
-                   (DT_LABEL(DT_PARENT(GPARENT(part)))), (_else_code))
-
-/* if 'jedec,spi-nor' return controller label or punt to _else_code */
-#define IS_JEDEC_SPI_NOR(part, _else_code) \
-       COND_CODE_1(DT_NODE_HAS_COMPAT(GPARENT(part), jedec_spi_nor), \
-                   (DT_LABEL(GPARENT(part))), (_else_code))
-
-/* if 'nordic,qspi-nor' return controller label or punt to _else_code */
-#define IS_NORDIC_QSPI_NOR(part, _else_code) \
-       COND_CODE_1(DT_NODE_HAS_COMPAT(GPARENT(part), nordic_qspi_nor), \
-                   (DT_LABEL(GPARENT(part))), (_else_code))
-
 /* return flash controller label based on matching compatible or NULL */
-#define DT_FLASH_DEV_FROM_PART(part) \
-       IS_SOC_NV_FLASH(part, \
-       IS_JEDEC_SPI_NOR(part, \
-       IS_NORDIC_QSPI_NOR(part, NULL)))
+#define DT_FLASH_DEV_FROM_PART(part)                                   \
+       DT_LABEL(COND_CODE_1(DT_NODE_HAS_COMPAT(GPARENT(part), soc_nv_flash), \
+                            (DT_PARENT(GPARENT(part))),                \
+                            (GPARENT(part))))
 
 #define FLASH_AREA_FOO(part)                           \
        {.fa_id = DT_FIXED_PARTITION_ID(part),          \
```

Reply:
The advantage of how it is currently done is that it is simple to add further special cases should they become necessary again. That said, I don't know how likely that is, and I don't have strong opinions one way or the other.

Reply:
My thought is that everything we've added has been a grandparent, except soc_nv_flash which is a great-grandparent.  If something else needs great-grandparent we can abstract the conditional with a `NEEDS_GGPARENT()`.  And whatever we get we need its label, so that can be lifted out to the top rather than repeated in each branch.

However we should get input from others; if the original approach is preferred I'll withdraw the NAK.

Reply:
@JordanYates I agree with @pabigot  here. Can you make the change so that we could proceed with the review?

Reply:
done

Reply:
Thanks!